### ISSUE TITLE
Style/MutableConstant doesn't offer to freeze a parenthesized expression containing non-primitive values

### ISSUE THREAD

**Issue by vittorius** [rubocop/rubocop#12066]
`Style/MutableConstant` cop doesn't offer to freeze a parenthesized expression in cases like:

```ruby
PARAMS_BASIC = %w[a b]
PARAMS_AUX = %w[c d]
COMBINED_PARAMS = (PARAMS_BASIC + PARAMS_AUX) # no demand to .freeze this
```

I had an assumption that "anything" that is right to the constant assignment that contains non-primitive values should be suggested to be frozen.

--------

## Expected behavior

```ruby
COMBINED_PARAMS = (PARAMS_BASIC + PARAMS_AUX).freeze
```

## Actual behavior

```ruby
COMBINED_PARAMS = (PARAMS_BASIC + PARAMS_AUX)
```

`rubocop --debug` output:

```
Inheriting configuration from { ... skimmed... }.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning { ... skimmed... }/freeze_parens_test.rb
.

1 file inspected, no offenses detected
Finished in 1.516961999997875 seconds
```

## Steps to reproduce the problem

See the intro text of this issue

## RuboCop version

```
% bundle exec rubocop -V
1.36.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.0.5) [arm64-darwin21]
```

Plugins' versions from Gemfile.lock (they are used though a corporate proxy gem so are not displayed in `rubocop -V` output):

```
rubocop-performance (1.13.3)
rubocop-rails (2.13.2)
rubocop-rake (0.6.0)
rubocop-rspec (2.8.0)
rubocop-thread_safety (0.4.4)
```

## Additional notes

Also, it would be great for this cop to detect cases of array addition/subtraction:

```ruby
A = [1, 2]
B = [2, 4]
C = A + B # turn this into C = (A + B).freeze
C = A - B # turn this into C = (A - B).freeze
```

But I realize that this could be impossible without tracking the types of the previously parsed expressions (I don't know the Rubocop internals, unfortunately).

**Comment by fatkodima** [comment]
This is expected behavior. You need to use `EnforcedStyle: strict` for this. See https://github.com/rubocop/rubocop/blob/3f8582b2f32a219e37caf9c93928e3cdeadefc4c/lib/rubocop/cop/style/mutable_constant.rb#L36-L81
